src/shadertools/lib/shader-cache.js

```diff
@@ -68,4 +72,40 @@ export default class ShaderCache {
     }
     return shader;
   }
+
+  // Retrive Program from cache if exists, otherwise create new instance.
+  getProgram(gl, {vs, fs}) {
+    assert(typeof vs === 'string', ERR_SOURCE);
+    assert(typeof fs === 'string', ERR_SOURCE);
+    const cacheKey = `VertexShader: ${vs} FragmentShader: ${fs}`;
```

Comment:
Just a thought: We might be duplicating a lot of long strings. An alternative approach could be a two level cache, that should at least allow for referencing the same strings.
```
const programCacheLevel1 = programCache[vs] || {};
const program = programCacheLevel2[fs] || null;
```